libs/user-identity-service/src/main/java/uk/gov/di/ipv/core/library/service/UserIdentityService.java

```diff
@@ -255,66 +230,61 @@ public boolean areVCsCorrelated(List<VcStoreItem> vcStoreItems)
         return true;
     }
 
-    private List<VcStoreItem> excludeTicfVC(List<VcStoreItem> vcStoreItems) {
-        return vcStoreItems.stream()
-                .filter(vcStoreItem -> !vcStoreItem.getCredentialIssuer().equals(TICF_CRI))
+    public List<VerifiableCredential> getIdentityCredentials(List<VerifiableCredential> vcs) {
+        return vcs.stream()
+                .filter(vcStoreItem -> !vcStoreItem.getCriId().equals(TICF_CRI))
                 .toList();
```

Comment:
I've removed getIdentityCredentials and in BuildCriOauthRequestHandler, I've modified this method to do it directly, since it's the only remaining use-case: 
```java
    private List<VerifiableCredential> getGpg45Vcs(String userId) throws CredentialParseException {
        return VcHelper.filterVCBasedOnProfileType(
                        verifiableCredentialService.getVcs(userId), ProfileType.GPG45)
                .stream()
                .filter(vc -> !vc.getCriId().equals(TICF_CRI))
                .toList();
    }
```

Reply:
I think you might need to be a little careful - for F2F we collect the TICF VC _before_ sending the user to the F2F CRI, so we might actually have a TICF VC in `BuildCriOauthRequestHandler`

Reply:
Ah yep, sorry I didn't explain that very well. It's still filtering out the TICF VC in BuildCriOauthRequestHandler